src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseCachingService.kt

```diff
@@ -0,0 +1,187 @@
+package nl.tudelft.ewi.se.ciselab.testgenie.services
+
+import com.intellij.openapi.diagnostic.Logger
+import org.evosuite.utils.CompactReport
+import org.evosuite.utils.CompactTestCase
+
+class TestCaseCachingService {
+    private val log: Logger = Logger.getInstance(this.javaClass)
+    private val files = mutableMapOf<String, FileTestCaseCache>()
+    private val filesLock = Object()
```

Comment:
Hey, how does this work. What is this object, what do you mean by lock? Thank you very much.

Reply:
This is a lock object to make sure that the `files` map can only be accessed by one thread at a time. Otherwise, we may run into concurrency issues.

Reply:
Okay, if you say so. Thanks